engine/src/main/java/org/hibernate/search/spi/IndexedTypesSet.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.spi;
+
+import java.util.Set;
+
+public interface IndexedTypesSet extends Iterable<IndexedTypeIdentifier> {
```

Comment:
there are deep thoughts here.. let's talk about it if your'e interested.

My whole point is to strictly avoid - or at least discourage - people to use their own collections to aggregate somehow a "type set" concept from multiple type-ids.

I definitely don't like throwing UnsupportedOperationException on the other methods, when there's no real reason to have a standard Set here.

By  exposing a very small API we can control how ti make it efficient.